index.html

```diff
@@ -2406,6 +2634,1425 @@ <h2>
+            </pre>
+            <p>
+              The [=remote end steps=] with command parameters |params| are:
+            </p>
+            <ol>
+              <li>
+                Let |simulatedGamepads| be the [=simulated gamepad map=].
+              </li>
+              <li>
+                If |simulatedGamepads|[|params|.|deviceId|] does not [=map/exist=], abort these steps.
+              </li>
+              <li>
+                Let |gamepad| be |simulatedGamepads|[|params|.|deviceId|].
+              </li>
+              <li>
+                Run the steps [=when a gamepad becomes unavailable on the system=] with |gamepad|.
+              </li>
+              <li>
+                [=map/Remove=] |simulatedGamepads|[|params|.|deviceId|].
+              </li>
+            </ol>
+          </section>
+        </section>
+        <section data-cite="infra">
+          <h4>
+            Events
+          </h4>
+          <section>
+            <h5>
+              The emulation.simulatedDeviceEvent Event
+            </h5>
+            <pre class="cddl">
+              emulation.SimulatedDeviceEvent = {
+                method: "emulation.removeSimulatedDevice",
+                params: emulation.SimulatedDeviceEventParams,
+              }
+            </pre>
+            <p>
+              To get <dfn>the identifier for a simulated gamepad</dfn> |gamepad|:
+            </p>
+            <ol>
+              <li>
+                Let |simulatedGamepads| be the [=simulated gamepad map=].
+              </li>
+              <li>
+                [=map/For each=] |key| → |value| of |simulatedGamepads|:
+                <ol>
+                  <li>
+                    If |value| is |gamepad|, return |key|.
+                  </li>
+                </ol>
+              </li>
+              <li>
+                Return `null`.
+              </li>
+            </ol>
+            <p>
+              To <dfn>vibrate a simulated gamepad</dfn> |gamepad| given a [=navigable=] |navigable| and {{GamepadEffectParameters}} |effectParams|:
+            </p>
+            <ol>
+              <li>
+                Let |deviceId:string| be [=the identifier for a simulated gamepad=] |gamepad|.
+              </li>
+              <li>
+                If |deviceId| is `null`, abort these steps.
+              </li>
+              <li>
+                Let |navigableId| be |navigable|'s [=navigable id=].
+              </li>
+              <li>
+                Let |eventParams| be a [=map=] matching the `emulation.SimulatedDeviceEventParameters` production with the `deviceId` field set to |deviceId|.
+              </li>
+              <li>
+                Let |vibration| be `false`.
+              </li>
+              <li>
+                Let |channels| be the [=list=] « `"strongMagnitude"`, `"weakMagnitude"`, `"leftTrigger"`, `"rightTrigger"` ».
+              </li>
+              <li>
+                [=list/For each=] |channel| of |channels|:
```

Comment:
It means that we ignore all other potential values. IDK if this is correct, and how extensible it is.